components/service-operator/README.md

```diff
@@ -1,15 +1,49 @@
 # Service Operator
 
+We aim for this to be our generic service operator, with a hope for it to be a
+provider agnostic one. The reason being, a desire to run the stack locally,
+without the worry of providing live database connections to a local deployment.
+
+## Adding new services
+
+Say you have a desire to add a new service to the mix. We're using a
+`kubebuilder` which essentially is our generator for all the things.
+
+Simply run:
+
+```sh
+kubebuilder create api --group database --version v1beta1 --kind MySQL
+...
+kubebuilder create api --group queue --version v1beta1 --kind RabbitMQ
+```
+
+Where `--group` defines the general nature of the service and `--kind` defines
+the engine we'd like to run.
+
+This will generate some `api/`, `config/` and `controller/` files, which you
+may want to look into. It usually comes with a nice set of comments, which will
+explain to you, what needs to be done in each section to get to your desired
+state.
+
+Note, that because it's essentially a generator, for most changes to the
+codebase, you may want to run `make generate`.
+
 ## PostgreSQL via AWS RDS
 
 Given a configuration file of the form:
 
 ```yaml
-  kind: Postgres
+---
+apiVersion: database.gsp.k8s.io/v1beta1
+kind: Postgres
+metadata:
+  name: postgres-sample
 spec:
   aws:
+    diskSizeGB: 100
```

Comment:
```suggestion
    diskSizeGb: 100
```

Reply:
Just noticed this changed in https://github.com/alphagov/gsp/pull/338/files#diff-52d4a1d56351dd58e13bb43410b0c27dR27. This is fine. I thought it was a typo.